client/elasticClient.go

```diff
@@ -323,6 +329,45 @@ func (ec *elasticClient) createAlias(alias string, index string) error {
 	return parseResponse(res, nil, elasticDefaultErrorResponseHandler)
 }
 
+func (ec *elasticClient) getWriteIndex(alias string) (string, error) {
+	res, err := ec.client.Indices.GetAlias(
+		ec.client.Indices.GetAlias.WithIndex(alias),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	var indexData map[string]struct {
+		Aliases map[string]struct {
+			IsWriteIndex bool `json:"is_write_index"`
+		} `json:"aliases"`
+	}
+
+	err = parseResponse(res, &indexData, elasticDefaultErrorResponseHandler)
+	if err != nil {
+		return "", err
+	}
+
+	// Iterate over the map and find the write index
+	var writeIndex string
+	for index, details := range indexData {
+		if len(indexData) == 1 {
```

Comment:
why not this if before the for loop and return 0, nil?

Reply:
because I don't know the key from the map